metals/src/main/scala/scala/meta/internal/metals/codeactions/CreateCompanionObjectCodeAction.scala

```diff
@@ -0,0 +1,134 @@
+package scala.meta.internal.metals.codeactions
+
+import org.eclipse.lsp4j.{CodeActionParams, Location}
+import org.eclipse.{lsp4j => l}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.meta.internal.metals.CodeAction
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+import scala.meta.Defn
+
+import scala.meta.internal.metals.ServerCommands
+
+/**
+ * look at the
+ *
+ * @param trees
+ */
+class CreateCompanionObjectCodeAction(
+    trees: Trees
+) extends CodeAction {
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  override def contribute(params: CodeActionParams, token: CancelToken)(implicit
+      ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    pprint.log("contribute")
+    val uri = params.getTextDocument().getUri()
+    val path = uri.toAbsolutePath
+    val range = params.getRange()
+    val applyTree =
+      if (range.getStart == range.getEnd)
+        trees
+          .findLastEnclosingAt[Defn.Class](
```

Comment:
We might want to do the same for trait also, in that case we might look for `Template` and check the parent.

Reply:
Yes! I would add that! Thanks for the tip!